Fix sass deprecation warnings

nmajor25 opened the pull request:
Shortcut Story ID: [sc-57510]

Main deprecation updates:

* Switch from deprecated `map-get()` to `map.get()`.
* Switch from using deprecated `@import`'s to `@use` & `@forward` instead.
  * More info on the deprecation [here](https://sass-lang.com/blog/import-is-deprecated/).

paulfalgout committed:
Remove subscribe getResource invoke

ws service supports models now.

paulfalgout committed:
Add workspace to all websocket subscriptions

Part of the filters work, this can currently always be added

paulfalgout committed:
Remove connectionResponseData

We didn’t ever use this.  Likely never will

paulfalgout committed:
Use sinon stub instead of wsIntercept helper

paulfalgout committed:
Only send data if data on socket open

if a socket just opens by default it’s not an issue.

paulfalgout committed:
Remove reopening sockets in the test environment

It reopens during changing tests

nmajor25 committed:
Small refactors on `'action attachments - show/hide icon in action list items'` test

paulfalgout committed:
Remove attaching resource and author to the notification payload

nmajor25 committed:
Remove unused `sass:color` import

nmajor25 committed:
Move `sass:math` import scope to file

Only used in the datepicker.scss file so we don't need to import it everywhere

nmajor25 committed:
Switch from deprecated `map-get()` => `map.get()`

nmajor25 committed:
Switch `@import`s to `@use` or `@forward`

nmajor25 commented:
I've got this PR set as a draft. I'd like to do more testing in my local dev environment. To make sure all the styles are working as before.

nmajor25 commented on vite.config.js:
```diff
@@ -32,7 +32,6 @@ const css = {
     scss: {
       additionalData: `
         @use 'sass:math';
-        @use 'sass:color';
```
I couldn't find anywhere this is being used.

nmajor25 commented on src/scss/core/_base.scss:
```diff
@@ -1,3 +1,5 @@
+@use '../provider-variables' as *;
```
The sass `preprocessorOptions` in the `vite.config.js` file adds `@use 'src/scss/provider-variables.scss' as *;` to the top of all our sass files? Or just the ones that are imported via javascript?

Because that didn't happen for these `/src/scss/core` sass files. I had to add these here for the `$` prefixed variables to work.

paulfalgout replied:
imports never needed this because they were effectively combined into the provider-core file which would have the include but `@forward` compiles and namespaces an export rather than putting each partial into the same namespace

nmajor25 replied:
Makes sense 👍

nmajor25 commented on src/scss/provider-variables.scss:
```diff
@@ -6,8 +6,10 @@
 //   Main Colors
 // -------------------------------------
 
+@use 'sass:map';
```
Eh this line should probably be placed above the `Main Color` comment...

paulfalgout commented on src/scss/provider-core.scss:
```diff
@@ -1,13 +1,13 @@
-@import 'core/space';
-@import 'core/text';
-@import 'core/typography';
+@forward 'core/reset';
```
So.. I'm not certain, but what we might be able to do here is:

```scss
@use '../provider-variables' as *;

@use 'core/reset';
...
@use 'core/typography';

@forward 'core/base';
...
@forward 'core/typography';
```

I think that would put them all in the same namespace and the export them all with the shared namespacing..  but I'm not 💯  on that.

I'm also not sure that it's better..

nmajor25 replied:
Hmm interesting, that's worth investigating. Not sure if that will work either, but would be nice if it did!

nmajor25 replied:
Couldn't get this to work 😞

paulfalgout replied:
one other thought that might be a hair cleaner.. at least for now.. we could make a js file at the very top of index called `provider-core-styles.js`.. remove this file and all the `_`d file names.. or something and instead of using sass partials just import all of these at the top..  then they'd get the `additionalData` and we wouldn't have to add that to the partials.

paulfalgout commented on vite.config.js:
```diff
@@ -31,9 +31,7 @@ const css = {
-        @use 'sass:math';
-        @use 'sass:color';
-        @import 'src/scss/provider-variables.scss';
+        @use 'src/scss/provider-variables.scss' as *;
```
It's possible that we should remove this altogether and break up the variables into separate `@use` namespaces imports.. at least that seems to be what scss wants us to do.. but.. I dunno.. this pattern of having a few global scss vars hasn't hurt us at all.. and the imports would be everywhere...

nmajor25 replied:
So to implement this we'd need to add `@use src/scss/provider-variables.scss` in every `.scss` file? With relative paths too? Like `@use ../../scss/provider-variables.scss`?

When working on this I thought about doing that. But I couldn't figure out another way besides having the `@use` in a ton of different files.

paulfalgout replied:
yeah.. well I think that's sass intention in deprecating `@import`.  They want explicit and obvious dependencies.. and for people to only use them when necessary... so they'd probably suggest against using `as *` and rather using `as colors` or something where the sass ends up `color: colors.blue;`   In theory I agree... but in practice.. yeah.. it's so many @use when our global "if you see a $variable, you'll find it in our short list of provider-variables.scss" seems to work pretty well for us.  The `additionalData` was always a little hacky.. but in some ways the more we fight the standard the more hacky our solutions are starting to look..   but as you said.. the alternative is just so much `@use` which doesn't really seem to clarify anything IMO.  So I dunno. sass sucks.

paulfalgout reviewed (commented):
Bah.. I dislike this change to sass..  but we gotta do something.. up for whichever TBH

nmajor25 committed:
Update @cypress/vite-dev-server, cypress-vite, vite-plugin-pwa, vite-plugin-stylelint npm packages

To support vite 6

nmajor25 committed:
On vite builds, convert yaml to js object ourselves instead of using external @modyfi/vite-plugin-yaml npm package

@modyfi/vite-plugin-yaml npm package is no longer being maintained and is blocking us from moving to vite v6

nmajor25 committed:
Upgrade to vite v6

Temporarily setting `css.preprocessorOptions.scss.api: 'legacy'` for vite 6 to work. We will remove that at a later date when we update sass to use the modern js api

nmajor25 committed:
Remove `scss/core/_overrides.scss` file

No longer needed

nmajor25 committed:
In our vite yaml plugin, use JSON.stringify() instead of tosource npm package

nmajor25 committed:
Add try...catch to our vite yaml plugin

nmajor25 committed:
Add `return null;`s to our vite plugins

nmajor25 commented:
Closing this PR and switching to a new one: https://github.com/RoundingWell/care-ops-frontend/pull/1382. I messed up a rebase.
